source/common/grpc/async_client_impl.h

```diff
@@ -292,20 +305,29 @@ class AsyncRequestImpl : public AsyncRequest,
   void onReceiveTrailingMetadata(Http::HeaderMapPtr&&) override {}
 
   void onRemoteClose(Grpc::Status::GrpcStatus status) override {
-    if (status != Grpc::Status::GrpcStatus::Ok) {
-      callbacks_.onFailure(status);
-      return;
+    Tracing::SpanFinalizerPtr finalizer = finalizer_factory_.create(request_, response_.get());
+    current_span_->setTag("grpc_status", std::to_string(status));
+
+    Grpc::Status::GrpcStatus reported_status = status;
+    if (status == Grpc::Status::GrpcStatus::Ok && response_ == nullptr) {
+      reported_status = Status::Internal;
     }
-    if (response_ == nullptr) {
-      callbacks_.onFailure(Status::Internal);
+
+    if (status != Grpc::Status::GrpcStatus::Ok) {
+      current_span_->setTag("error", "true");
+      current_span_->finishSpan(*finalizer);
+      callbacks_.onFailure(reported_status);
```

Comment:
I was wondering if the success/failure callbacks might want to decorate the span (or be included in the timing that the span covers). I don't think they have access to the current span though. Thoughts here?